extensions/wikidata/src/org/openrefine/wikidata/exporters/QSValuePrinter.java

```
@@ -0,0 +1,106 @@
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ ******************************************************************************/
+package org.openrefine.wikidata.exporters;
+
+import java.math.BigDecimal;
+import java.util.Locale;
+
+import org.openrefine.wikidata.schema.entityvalues.ReconEntityIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.DatatypeIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.EntityIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.GlobeCoordinatesValue;
+import org.wikidata.wdtk.datamodel.interfaces.MonolingualTextValue;
+import org.wikidata.wdtk.datamodel.interfaces.QuantityValue;
+import org.wikidata.wdtk.datamodel.interfaces.StringValue;
+import org.wikidata.wdtk.datamodel.interfaces.TimeValue;
+import org.wikidata.wdtk.datamodel.interfaces.ValueVisitor;
+
+/**
+ * Prints a Wikibase value as a string as required by QuickStatements. Format
+ * documentation: https://www.wikidata.org/wiki/Help:QuickStatements
+ * 
+ * Any new entity id will be assumed to be the last one created, represented
+ * with "LAST". It is fine to do this assumption because we are working on edit
+ * batches previously scheduled by {@link QuickStatementsUpdateScheduler}.
+ * 
+ * @author Antonin Delpeuch
+ *
+ */
+public class QSValuePrinter implements ValueVisitor<String> {
+
+    @Override
+    public String visit(DatatypeIdValue value) {
```

A unsupported exception should be thrown instead of return null silently